### PR TITLE
Ac/projects

### PR DESCRIPTION
**Pull request by croysdaa**
This PR adds changes to the add project and room pages, the edit project page, the project list page and the room list page.
addprojectpage
addroompage
editprojectpage
projectslistpage
projectsroomlistpage

### CODE REVIEW

File: back-end/src/main/java/app/cluttermap/controller/ProjectsController.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+import app.cluttermap.model.Project;
+import app.cluttermap.model.Room;
+import app.cluttermap.repository.ProjectsRepository;
+import app.cluttermap.repository.RoomsRepository;
+
+@RestController
+@RequestMapping("/projects")
+@CrossOrigin(origins = "http://localhost:3000")
```
Comment (croysdar) [inline back-end/src/main/java/app/cluttermap/controller/ProjectsController.java:26]: I think you can remove this line after you merge main into this branch.

File: back-end/src/main/java/app/cluttermap/controller/RoomsController.java [hunk 2]
```diff
@@ -35,13 +42,24 @@ public Iterable<Room> getRooms() {
     }
 
     @PostMapping()
-    public ResponseEntity<Room> addOneRoom(@RequestBody Room room) {
-        if (room.getName() == null || room.getName().isEmpty()) {
+    public ResponseEntity<Room> addOneRoom(@RequestBody NewRoomDTO roomDTO) {
+        if (roomDTO.getName() == null || roomDTO.getName().isEmpty()) {
```
Comment (croysdar) [inline back-end/src/main/java/app/cluttermap/controller/RoomsController.java:44]: Might as well combine these two conditionals

File: back-end/src/main/java/app/cluttermap/dto/NewRoomDTO.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package app.cluttermap.dto;
+
+public class NewRoomDTO {
```
Comment (croysdar) [inline back-end/src/main/java/app/cluttermap/dto/NewRoomDTO.java:3]: maybe add a comment explaining what this is for
DTO = data transfer object

File: back-end/src/main/java/app/cluttermap/model/Project.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        this.rooms = rooms;
+    }
+
+    // Utility methods for convenience
+    public void addRoom(Room room) {
+        rooms.add(room);
+        room.setProject(this);
+    }
+
+    public void removeRoom(Room room) {
```
Comment (croysdar) [inline back-end/src/main/java/app/cluttermap/model/Project.java:70]: You should probably use this in the RoomsController delete function, similar to how you used the above addRoom in the RoomsController

File: react-front-end/src/features/projects/ProjectMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            >
+                {/* 
+                Edit Project - takes you to an edit page?
+                Add location - takes you to 'new location' page?
+            */}
+                <MenuItem onClick={handleEdit}>
+                    Edit Project
+                </MenuItem>
+                <MenuItem onClick={handleAddLocation}>
+                    Add Location
```
Comment (croysdar) [inline react-front-end/src/features/projects/ProjectMenu.tsx:66]: I don't think you need Add Location in this file at all

File: react-front-end/src/features/projects/ProjectsList.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+    if (isLoading) {
+        return <div>Loading...</div>;
+    }
+
+    if (isError) {
+        return <div>{error.toString()}</div>
+    }
+
+    const renderLocation = (location: Location) => {
```
Comment (croysdar) [inline react-front-end/src/features/projects/ProjectsList.tsx:53]: You can remove this function

File: react-front-end/src/features/rooms/AddRoom.tsx [hunk 7]
```diff
@@ -89,6 +98,17 @@ export const AddRoom = () => {
                         </Button>
                     </form>
 
+                    <Button
```
Comment (croysdar) [inline react-front-end/src/features/rooms/AddRoom.tsx:101]: I think the spacing here is off, should be tabbed over one. And remove the extra whitespace line as well

File: back-end/src/main/java/app/cluttermap/model/Project.java (hunk 4 above)
Comment (croysdaa) [reply back-end/src/main/java/app/cluttermap/model/Project.java:70]: I don't see addRoom used in RoomsController

**Review by croysdar — APPROVED** [review]

**Pull request merged**